### ISSUE TITLE
(bench/riker) failure on redis compilation 

### ISSUE THREAD

**Issue by ezrizhu** [binpash/hs#92]
redis spec-run exits fine but didn't generate files, sh works, running against the rikerfile provided by riker repo, on the specified redis commit
```
docker run --privileged --rm -it ezri/hs /bin/bash
# inside the container
mkdir /root/redis
cd /root/redis
git clone https://github.com/redis/redis.git .
git checkout d96f47cf06b1cc24b82109e0e87ac5428517525a
curl https://raw.githubusercontent.com/curtsinger-lab/riker/main/benchmarks/redis/files/Rikerfile > run.sh
make # need to resolve a few deps it seems
make clean
sh run.sh # this works
file src/redis-server # verif
make clean # cleanup
/srv/hs/pash-spec.sh -d2 run.sh
file src/redis-server # verif (this fails)
```

log: https://bns.sh/b.txt (expires in 30d)

(tested on fix-conditional)

**Comment by ezrizhu** [comment]
Issue still persists with latest main 13fc64ed3d3f696959f34d22a3d6f04b7178e71a
https://bns.sh/T.txt

**Comment by ezrizhu** [comment]
appears to be fixed by #93 
will double check

**Comment by ezrizhu** [comment]
build works with regular shell
```
docker run --privileged --rm ghcr.io/binpash/hs/redis:latest /bin/bash -c 'time sh /root/redis/redis_build.sh && file /root/redis/src/redis-server'
```

```
real    0m14.515s
user    0m13.439s
sys     0m0.800s
/root/redis/src/redis-server: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=e4bf8517ffe0c4edf32515ca936b6be5dc22a034, for GNU/Linux 3.2.0, with debug_info, not stripped
```

build fails with hs
```
docker run --privileged --rm ghcr.io/binpash/hs/redis:latest /bin/bash -c 'time /srv/hs/pash-spec.sh /root/redis/redis_build.sh && file /root/redis/src/redis-server'
```

```
Traceback (most recent call last):
  File "/srv/hs/parallel-orch/scheduler_server.py", line 240, in <module>                                                                                                                                                                  main()
  File "/srv/hs/parallel-orch/scheduler_server.py", line 236, in main
    scheduler.run()
  File "/srv/hs/parallel-orch/scheduler_server.py", line 197, in run
    self.partial_program_order.eager_fs_killing()
  File "/srv/hs/parallel-orch/partial_program_order.py", line 557, in eager_fs_killing
    self.fetch_fs_actions()
  File "/srv/hs/parallel-orch/partial_program_order.py", line 401, in fetch_fs_actions
    node.gather_fs_actions()
  File "/srv/hs/parallel-orch/node.py", line 484, in gather_fs_actions
    read_set, write_set = trace_v2.parse_and_gather_cmd_rw_sets(trace_object)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/hs/parallel-orch/trace_v2.py", line 375, in parse_and_gather_cmd_rw_sets
    all_records = [r for record in records for r in record.closure()]
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/hs/parallel-orch/trace_v2.py", line 375, in <listcomp>
    all_records = [r for record in records for r in record.closure()]
                                                    ^^^^^^^^^^^^^^
AttributeError: 'tuple' object has no attribute 'closure'
```

workaround: appears to be resolved by making some deps?
```
docker run --privileged --rm ghcr.io/binpash/hs/redis:latest /bin/bash -c 'cd /root/redis && make &>/dev/null &&make clean && time sh /srv/hs/pash-spec.sh /root/redis/redis_build.sh && file /root/redis/src/redis-server'
```

```
real    0m24.309s
user    0m20.368s
sys     0m3.903s
/root/redis/src/redis-server: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=36a1f18dbce7bf35eafe05da751f2d6906578341, for GNU/Linux 3.2.0, with debug_info, not stripped
```

**Comment by ezrizhu** [comment]
resolved via ebff1d4234e77c0dadd82e4c0463250a3d704381

```
docker run --privileged --rm ghcr.io/binpash/hs/redis:latest /bin/bash -c 'time /srv/hs/pash-spec.sh /root/redis/redis_build.sh && file /root/redis/src/redis-server'
```

```
real    0m24.419s
user    0m20.199s
sys     0m3.989s
/root/redis/src/redis-server: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=742ccbb3ac266d96037789d2d1d98e2e2f3649f4, for GNU/Linux 3.2.0, with debug_info, not stripped
```
